src/Avalonia.Controls/Flyouts/FlyoutBase.cs

```diff
@@ -0,0 +1,487 @@
+
+        /// <summary>
+        /// Defines the AttachedFlyout property
+        /// </summary>
+        public static readonly AttachedProperty<FlyoutBase?> AttachedFlyoutProperty =
+            AvaloniaProperty.RegisterAttached<FlyoutBase, Control, FlyoutBase?>("AttachedFlyout", null);
+
+        private bool _isOpen;
+        private Control? _target;
+        protected Popup? _popup;
+        private FlyoutShowMode _showMode = FlyoutShowMode.Standard;
+        Rect? enlargedPopupRect;
+        IDisposable? transientDisposable;
+
+        /// <summary>
+        /// Gets whether this Flyout is currently Open
+        /// </summary>
+        public bool IsOpen
+        {
+            get => _isOpen;
+            private set => SetAndRaise(IsOpenProperty, ref _isOpen, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the desired placement
+        /// </summary>
+        public FlyoutPlacementMode Placement
+        {
+            get => GetValue(PlacementProperty);
+            set => SetValue(PlacementProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the desired ShowMode
+        /// </summary>
+        public FlyoutShowMode ShowMode
+        {
+            get => _showMode;
+            set => SetAndRaise(ShowModeProperty, ref _showMode, value);
+        }
+
+        /// <summary>
+        /// Gets the Target used for showing the Flyout
+        /// </summary>
+        public Control? Target
+        {
+            get => _target;
+            private set => SetAndRaise(TargetProperty, ref _target, value);
+        }
+
+        public event EventHandler? Closed;
+        public event EventHandler<CancelEventArgs>? Closing;
+        public event EventHandler? Opened;
+        public event EventHandler? Opening;
+
+        public static FlyoutBase? GetAttachedFlyout(Control element)
+        {
+            return element.GetValue(AttachedFlyoutProperty);
+        }
+
+        public static void SetAttachedFlyout(Control element, FlyoutBase? value)
+        {
+            element.SetValue(AttachedFlyoutProperty, value);
+        }
+
+        public static void ShowAttachedFlyout(Control flyoutOwner)
+        {
+            var flyout = GetAttachedFlyout(flyoutOwner);
+            flyout?.ShowAt(flyoutOwner);
+        }
+
+        /// <summary>
+        /// Shows the Flyout at the given Control
+        /// </summary>
+        /// <param name="placementTarget">The control to show the Flyout at</param>
+        public void ShowAt(Control placementTarget)
+        {
+            ShowAtCore(placementTarget);
+        }
+
+        /// <summary>
+        /// Shows the Flyout for the given control at the current pointer location, as in a ContextFlyout
+        /// </summary>
+        /// <param name="placementTarget">The target control</param>
+        /// <param name="showAtPointer">True to show at pointer</param>
+        public void ShowAt(Control placementTarget, bool showAtPointer)
```

Comment:
It's probably bad idea to make popup find out by itself where is the pointer located instead of telling him specific location as it was done in UWP. Especially when it should be pretty straightforward to pass pointer location from PointerPressed event to this method.
But it's not something that can be achieved in this PR without big changes in Popup positioning code (I haven't checked it detailed though).
@kekekeks

Reply:
So for now I think it's fine to have this method like it is and deprecated later, when Popup will be able to be positioned on specific pointer position.